cmd/monaco/v2/deploy/cmd.go

```diff
@@ -0,0 +1,197 @@
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/rest"
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/util/client"
+	"github.com/spf13/afero"
+)
+
+func Deploy(fs afero.Fs, deploymentManifestPath string, specificEnvironment string,
+	specificProject string, dryRun, continueOnError bool) error {
+
+	deploymentManifestPath = filepath.Clean(deploymentManifestPath)
+
+	manifest, errs := manifest.LoadManifest(fs, deploymentManifestPath)
+
+	if errs != nil {
+		for _, err := range errs {
+			fmt.Println(err)
+		}
+		return errors.New("error while loading environments")
+	}
+
+	environments := manifest.GetEnvironmentsAsSlice()
+
+	if specificEnvironment != "" {
+		filtered, err := filterEnvironmentByName(environments, specificEnvironment)
+
+		if err != nil {
+			return err
+		}
+
+		environments = filtered
+	}
+
+	environmentMap := toEnvironmentMap(environments)
+
+	workingDir := filepath.Dir(deploymentManifestPath)
+	apis := api.NewApis()
+
+	projects, errs := project.LoadProjects(fs, project.ProjectLoaderContext{
+		Apis:            getApiNames(apis),
+		WorkingDir:      workingDir,
+		Manifest:        manifest,
+		ParametersSerde: config.DefaultParameterParsers,
+	})
+
+	if errs != nil {
+		return errors.New("error while loading projects")
+	}
+
+	if specificProject != "" {
+		filtered, err := filterProjectsByName(projects, specificProject)
+
+		if err != nil {
+			return err
+		}
+
+		projects = filtered
+	}
+
+	sortedConfigs, errs := sort.GetSortedConfigsForEnvironments(projects, toEnvironmentNames(environments))
+
+	if errs != nil {
+		return errors.New("error during sort")
+	}
+
+	var deploymentErrors []error
+
+	for envName, configs := range sortedConfigs {
+		println("deploy env: ", envName)
+		env, found := environmentMap[envName]
+
+		if !found {
+			if continueOnError {
+				deploymentErrors = append(deploymentErrors, fmt.Errorf("cannot find environment `%s`", envName))
+				continue
+			} else {
+				return fmt.Errorf("cannot find environment `%s`", envName)
+			}
+		}
+
+		client, err := getClient(env, dryRun)
+
+		if err != nil {
+			if continueOnError {
+				deploymentErrors = append(deploymentErrors, err)
+				continue
+			} else {
+				return err
+			}
+		}
+
+		errors := deploy.DeployConfigs(client, apis, configs, continueOnError, dryRun)
+
+		if errors != nil {
+			deploymentErrors = append(deploymentErrors, errors...)
+		}
+	}
+
+	if deploymentErrors != nil {
+		for _, err := range deploymentErrors {
+			fmt.Println(err)
+		}
+
+		return errors.New("errors during deploy")
+	}
+
+	return nil
+}
+
+func toEnvironmentNames(environments []manifest.EnvironmentDefinition) []string {
+	result := make([]string, 0, len(environments))
+
+	for _, env := range environments {
+		result = append(result, env.Name)
+	}
+
+	return result
+}
+
+func filterProjectsByName(projects []project.Project, name string) ([]project.Project, error) {
+	var result []project.Project
+
+	for _, p := range projects {
+		if p.Id == name {
```

Comment:
You drop all projects that the 'specific project' has dependencies on don't you?
Actually not sure if we currently do this, but I don't think we should. 
Deploying a project without it's dependencies is useless - either we go for ease of use and deploy needed projects, or fail with an error and force you to list all

Reply:
Yeah I noticed that as well :laughing: 
Also specific projects only allow for one project now, not multiple :laughing:

Reply:
You already have the list of Dependencies of a Project at this point right? 
If you find the specific project and then re-add everything that it depends on you're good (can't currently think of any way you could to do that in one pass, as you need the Project and initially have just the name)

Reply:
Should be fixed :+1: